src/user/controller/item.ts

```diff
@@ -52,12 +52,16 @@ class UserController extends Controller {
     const principalService = new PrincipalService(ctx.privileges);
 
     const user = await principalService.findByExternalId(ctx.params.id);
-    user.active = !!ctx.request.body.active;
-    user.nickname = ctx.request.body.nickname;
+  
+    if (user.type === 'user') {
+      user.active = !!ctx.request.body.active;
+      user.nickname = ctx.request.body.nickname;
 
+      await userService.updateUserInfo(user, ctx.request.body.userInfo);
+    }
```

Comment:
RE:

> Check failure on line 60 in src/user/controller/item.ts
> 
> GitHub Actions
> / Build
> 
> Property 'userInfo' does not exist on type 'PrincipalEdit'.

I didn't want to go modifying more stuff if this wasn't the right place to do that

Reply:
So there's a bit of a tricky thing here. Right now the `put()` action on resources supports both HTML forms and PUT requests using `appllication/json`, but with this more complex structure i think we need a different schema for each of these cases.

It's needed because if we don't, we don't have strict validation for any of this and that feels like a bad idea.

So i think I will spend some time tonight fixing this and I'll commit this directly to this branch.

In the end we'll have 2 JSON schemas, one for JSON and one for forms. So stay tuned

Reply:
you mean we need to have a JSON schema for `application/json`, and `multipart/form-data` right?

Yeah I agree with that

Reply:
Yeah exactly. They're so different now. We could get away with having 1 before